browser/ui/webui/settings/brave_appearance_handler.cc

```diff
@@ -179,6 +188,22 @@ void BraveAppearanceHandler::GetIsGeminiSupported(
   ResolveJavascriptCallback(args->GetList()[0], base::Value(is_supported));
 }
 
+void BraveAppearanceHandler::GetIsBitcoinDotComSupported(
+    const base::ListValue* args) {
+  CHECK_EQ(args->GetSize(), 1U);
+
+  AllowJavascript();
+
+#if !BUILDFLAG(MOONPAY_ENABLED)
```

Comment:
I feel like the BraveAppearanceHandler shouldn't have to know that moonpay is supported based on `ntp_widget_utils::IsRegionSupported` and also what happens if moonpay is enabled and `ntp_widget_utils_enabled` is not? If that is not a valid configuration then you should add `if (moonpay_enabled) { asssert(ntp_widget_utils_enabled) }` in the gn config

Reply:
after our discussion this makes more sense and this is following the pattern from binance, etc... so don't worry about it for now, but ideally we should try to avoid exposing the buildflags to the calling code. So for instance we might have `moonpay_utils::IsRegionSupported` which is not inside `MOONPAY_ENABLED` buildflag and then it looks something like this
```
bool IsRegionSupported(PrefService* prefs) {
#if !BUILDFLAG(MOONPAY_ENABLED)
  return false;
#else 
  return ntp_widget_utils::IsRegionSupported(
      prefs, moonpay::bitcoin_dot_com_supported_regions, true);
#endif 
}

Reply:
I'm not sure why any of this is in the `BraveAppearanceHandler` when this isn't in the _Appearance_ section. It's in the _New Tab Page_ section https://github.com/brave/brave-core/pull/6681/files#diff-0f69784376867456eadc3fde6ea4f39cR80

Reply:
to be followed up